### DISCUSSION TITLE
Is it possible to print your whole website to PDFs?

### DISCUSSION THREAD

**Discussion by IceManTBM3** [quarto-dev/quarto-cli#5782] [Q&A]
### Description

We publish a HTML website, a desire is to print the site to a single or multiple PDF(s) is this doable?

I know you can go to each file/page and set the output type, but what would be really desirable would be to run some quarto render command, is this possible?

I don't think so, but just thought to ask.

**Comment by mcanouil** [comment]
Indeed, you can't render a website as PDF.
Only render each documents as PDF.

**Reply by cscheid** [reply]
What @mcanouil is accurate.

With that said, if you use the "book" project type, then the HTML format is a "book website" with the chapter contents in separate files, while the "PDF" format is a single PDF file. See https://quarto.org/docs/books/index.html for more.

**Reply by mcanouil** [reply]
If you tweak the navbar/sidebar, you can have something close to a "real" website.
This could be a good trick to get a "website" as PDF.

**Reply by IceManTBM3** [reply]
Thanks guys....

**Comment by cderv** [comment]
This would be one use case of integrating with Paged.js 

It is not trivial though:

* It would require build a single HTML page version of the website 
* Having a CSS print / paged media that would organized the content in a paginated way. 
* Loading and running the paged.js tooling to create a paginated HTML version of the website.
* And then print to PDF with chrome. 

Not trivial, but possible IMO. Just some ideas and thoughts that I have in mind since some time now. Way more suited for a book output than a website though... and there is now the [typst](https://quarto.org/docs/prerelease/1.4/typst.html) format also but not for book yet I believe. 

Anyhow, fun project! :)

**Comment by nrennie** [comment]
Just adding this link here for anyone who's also trying to create a PDF from a Quarto website (not a book website) since I came across this discussion when searching for solutions myself. This is the workaround solution we came up with: https://github.com/royal-statistical-society/datavisguide/blob/main/_make_pdf.qmd

We create another .qmd file with an R function that strips out the YAML from each webpage .qmd and fixes any image file paths, manually specify the order of pages, then render to PDF. Perhaps not a great solution for larger, more complex sites but this worked fine for our use case.

**Reply by mcanouil** [reply]
Are you using conditional content and/or listing in the website you turn into a pdf? 🤔 (just wondering how far this approach can work)

**Reply by nrennie** [reply]
Yes, there's a few bits of conditional content in the website - I think just a couple of images that are "don't display if output is PDF"

**Reply by DOH-FAA3303** [reply]
@nrennie this is so useful, thank you for sharing! I hope quarto devs see this and expand on it

**Reply by humoroussmile** [reply]
This is an awesome workaround! Thank you!!

**Comment by ptram** [comment]
Unfortunately, it seems at the moment the book format is not supported with typst. At least, so the Quarto 1.4 compiler says when I try.

I'm trying to do the same thing as the OP. Ahat I'm trying to do is reuse the navigation structure, contained in the _quarto.yml generated for a website, for the book. Apparently, it is a matter of converting sections to parts, and content files to chapters. I guess the structure should also be simplified a bit (does the book allows more than three levels of nesting?), but I'm still exploring.
